azure-kusto-data/azure/kusto/data/_converters.py

```diff
@@ -3,19 +3,49 @@
 from datetime import timedelta
 import re
 from dateutil import parser
+from dateutil.tz import UTC
 import six
 
 # Regex for TimeSpan
 _TIMESPAN_PATTERN = re.compile(r"(-?)((?P<d>[0-9]*).)?(?P<h>[0-9]{2}):(?P<m>[0-9]{2}):(?P<s>[0-9]{2}(\.[0-9]+)?$)")
 
 
+def to_high_precision_type(kusto_type, raw_value, typed_value):
```

Comment:
Maybe we want to locate this after the vanilla python converters? It will be less used.
Do you want to separate this method into two, one that takes timedelta and the other datetime?
This way every method will be consisted in its return type, and it will be easier providing names to those methods